### PR TITLE
foedag main window title fixed

### PR DESCRIPTION
**Pull request by zaryabkhan-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
Foedag's Main Window title was showed in this pattern:  "FOEDAG - current opened file path".
image

> #### What is currently done? (Provide issue link if applicable)
Now main window title is changed to following consistent pattern:  "current-file-open-in-editor" - "project-name" - FOEDAG.
>
> #### What does this pull request change?
It will change the title pattern of main window. 

> #### Testing Steps:
Run the Gui and create or open new project.
Check the title on the window after creating or opening project
Main title shows in the following format shows in the image below
image

> ### Which part of the code base require a change
> 
> - [x] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [x] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/MainWindow/main_window.cpp [hunk 1]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
 }
 
 void MainWindow::newDesignCreated(const QString& design) {
-  setWindowTitle(m_projectInfo.name + " - " + design);
+  QFileInfo path(design);
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:180]: const here

File: src/MainWindow/main_window.cpp [hunk 2]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
...
+  QString strName = path.baseName();
+  QString fileName = path.fileName();
+  qDebug() << "strNme = " << strName;
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:183]: why it here

File: src/MainWindow/main_window.cpp [hunk 3]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
...
+  if (strName == "") {
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:184]: please use isEmpty()

File: src/MainWindow/main_window.cpp [hunk 4]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
 }
 
 void MainWindow::newDesignCreated(const QString& design) {
-  setWindowTitle(m_projectInfo.name + " - " + design);
+  QFileInfo path(design);
+  QString strName = path.baseName();
+  QString fileName = path.fileName();
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:182]: remove this

File: src/MainWindow/main_window.cpp [hunk 5]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
...
+  qDebug() << "strNme = " << strName;
+  if (strName == "") {
+    setWindowTitle(m_projectInfo.name + "-");
```
[the remark this answers is not in the document]
Comment (zaryabkhan-rs) [reply src/MainWindow/main_window.cpp:185]: it was as it is before.

File: src/MainWindow/main_window.cpp [hunk 6]
```diff
@@ -177,7 +177,15 @@ void MainWindow::openFileSlot() {
...
+  } else {
+    setWindowTitle(fileName + " - " + strName + " - " + m_projectInfo.name);
```
Comment (KochynVolodymyr) [inline src/MainWindow/main_window.cpp:187]: const QFileInfo path(design);
setWindowTitle(path.fileName() + " - " + path.baseName() + " - " + m_projectInfo.name);

Comment (zaryabkhan-rs) [reply src/MainWindow/main_window.cpp]: Yeah I have fixed like this now.

**Comment by KochynVolodymyr** [comment]
This implementation doesn't meat the requirements.

**Pull request closed without merging**